diff --git a/images/Makefile.common.in b/images/Makefile.common.in
index c67ae50..7d73921 100644
--- a/images/Makefile.common.in
+++ b/images/Makefile.common.in
@@ -17,18 +17,18 @@
 # get image name from directory we're building
 IMAGE_NAME?=$(notdir $(CURDIR))
 # docker image registry, default to upstream
-REGISTRY?=gcr.io/k8s-staging-kind
+REGISTRY?=kindest
 # for appending build-meta like "_containerd-v1.7.1"
 TAG_SUFFIX?=
 # tag based on date-sha
-TAG?=$(shell echo "$$(date +v%Y%m%d)-$$(git log --pretty=format:'%h' -n 1)")
+TAG?=v20260601-995e8fa5
 # the full image tag
-IMAGE?=$(REGISTRY)/$(IMAGE_NAME):$(TAG)$(TAG_SUFFIX)
+IMAGE?=$(REGISTRY)/$(IMAGE_NAME):$(TAG)
 # Go version to use, respected by images that build go binaries
 GO_VERSION=$(shell cat $(CURDIR)/../../.go-version | head -n1)
 
 # build with buildx
-PLATFORMS?=linux/amd64,linux/arm64
+PLATFORMS?=linux/s390x
 OUTPUT?=
 PROGRESS=auto
 EXTRA_BUILD_OPT?=
diff --git a/images/local-path-helper/stage-binary-and-deps.sh b/images/local-path-helper/stage-binary-and-deps.sh
index 88fac24..8c6b8a8 100755
--- a/images/local-path-helper/stage-binary-and-deps.sh
+++ b/images/local-path-helper/stage-binary-and-deps.sh
@@ -78,7 +78,7 @@ binary_to_libraries() {
     | awk '{print $1}' \
     `# linux-vdso.so.1 is a special virtual shared object from the kernel` \
     `# see: http://man7.org/linux/man-pages/man7/vdso.7.html` \
-    | grep -v 'linux-vdso.so.1'
+    | grep -v -e 'linux-vdso.so.1' -e 'linux-vdso64.so.1'
 }
 
 # main script logic
diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index 192e1ac..49f1dfc 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -110,10 +110,9 @@ COPY --chmod=0755 scripts/third_party/gimme/gimme /usr/local/bin/
 COPY --chmod=0755 scripts/target-cc /usr/local/bin/
 # tools needed at build-time only
 # first ensure we can install packages for both architectures
-RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 \
+RUN dpkg --add-architecture s390x \
     && clean-install bash ca-certificates curl git make pkg-config \
-    build-essential crossbuild-essential-amd64 crossbuild-essential-arm64 \
-    libseccomp-dev:amd64 libseccomp-dev:arm64
+    build-essential libseccomp-dev
 # set by makefile to .go-version
 ARG GO_VERSION
 RUN eval "$(gimme "${GO_VERSION}")" \
diff --git a/pkg/cluster/internal/kubeadm/config.go b/pkg/cluster/internal/kubeadm/config.go
index 6cf1f9c..1176fe4 100644
--- a/pkg/cluster/internal/kubeadm/config.go
+++ b/pkg/cluster/internal/kubeadm/config.go
@@ -600,6 +600,7 @@ metadata:
 cgroupDriver: {{ .CgroupDriver }}
 cgroupRoot: /kubelet
 failSwapOn: false
+failCgroupV1: false
 # configure ipv6 addresses in IPv6 mode
 {{ if .IPv6 -}}
 address: "::"
diff --git a/pkg/build/nodeimage/const_storage.go b/pkg/build/nodeimage/const_storage.go
index 2d1db53..87101d0 100644
--- a/pkg/build/nodeimage/const_storage.go
+++ b/pkg/build/nodeimage/const_storage.go
@@ -20,13 +20,13 @@ package nodeimage
 The default PV driver manifest and images are provisionally rancher.io/local-path-provisioner
 NOTE: we have customized it in the following ways:
 - storage is under /var instead of /opt
-- our own image and helper image
+- upstream local-path-provisioner image and our own helper image
 - schedule to linux nodes only
 - install as the default storage class
 - tolerate control plane scheduling taints
 */
 
-const storageProvisionerImage = "docker.io/kindest/local-path-provisioner:v20260521-9fb22683"
+const storageProvisionerImage = "docker.io/rancher/local-path-provisioner:v0.0.34"
 const storageHelperImage = "docker.io/kindest/local-path-helper:v20260131-7181c60a"
 
 // image we need to preload
